packages/features/data-table/components/useFetchMoreOnBottomReached.ts

```diff
@@ -2,23 +2,27 @@
 
 import { useCallback, useEffect } from "react";
 
-export const useFetchMoreOnBottomReached = (
-  tableContainerRef: React.RefObject<HTMLDivElement>,
-  fetchNextPage: () => void,
-  isFetching: boolean,
-  totalFetched: number,
-  totalDBRowCount: number
-) => {
+export const useFetchMoreOnBottomReached = ({
+  tableContainerRef,
+  hasNextPage,
+  fetchNextPage,
+  isFetching,
+}: {
+  tableContainerRef: React.RefObject<HTMLDivElement>;
+  hasNextPage: boolean;
+  fetchNextPage: () => void;
+  isFetching: boolean;
+}) => {
   const fetchMoreOnBottomReached = useCallback(
     (containerRefElement?: HTMLDivElement | null) => {
       if (containerRefElement) {
         const { scrollHeight, scrollTop, clientHeight } = containerRefElement;
-        if (scrollHeight - scrollTop - clientHeight < 300 && !isFetching && totalFetched < totalDBRowCount) {
+        if (scrollHeight - scrollTop - clientHeight < 300 && !isFetching && hasNextPage) {
           fetchNextPage();
         }
       }
     },
-    [fetchNextPage, isFetching, totalFetched, totalDBRowCount]
+    [fetchNextPage, isFetching]
```

Comment:
We relied on `totalFetched` and `totalDBRowCount`. However, when filters are applied, we cannot.

For example, let's say the filter result is only a single row. Then it's like:

```
totalFetched === 1
totalDBRowCount === 23424 // whatever the total db row count
```

Then this if condition is true, so it tries to fetch the next page even though there is no next page at all. So from now on, we use `hasNextPage` variable instead.